embedded/store/immustore.go

```diff
@@ -1118,7 +1118,6 @@ func (s *ImmuStore) precommit(otx *OngoingTx, expectedHeader *TxHeader, waitForI
 
 	if expectedHeader == nil {
 		ts = s.timeFunc().Unix()
-		blTxID = s.aht.Size()
```

Comment:
it may be better to leave only the header version out of the critical section. Moving timestamp and validations during replicated commit inside it, to prevent potential edge cases when switching between master and replica modes. Please take a look at the commit in branch `fix/bltx-ordering-rep` as reference

Reply:
or situation where the same tx is attempted to be replicated multiple times at the same time. The disadvantage is that more garbage in payload logs may be generated in such situations

Reply:
I have moved the timestamp calculation to the locked section - also improved the test to ensure timestamps are ordered.

Also added check for simultaneous replicators - currently the check is done both before the mutex is held (like checking if we have the data or that transaction hash is correct) and while the mutex is locked (final check to ensure there was no additional TX in the meantime).